gitlab.go

```diff
@@ -187,9 +187,6 @@ func (c *Client) SetBaseURL(urlStr string) error {
 		return err
 	}
 
-	// Set the encoded opaque data
-	c.baseURL.Opaque = c.baseURL.Path
-
```

Comment:
After a good night sleep, I decided I would like to hold of moving to Go 1.5 as a minimum requirement for this package. As we don't know who is using this package and what versions of Go they might be using, this could potentially break software using this package.

Next to that doesn't change or fix anything other then that to code is slightly cleaner. So let's wait with changing this to something that requires at least Go 1.5 for now...